docs/input-manifest/api-reference.md

```diff
@@ -29,167 +29,139 @@ Manifest is a definition of the user's infrastructure. It contains cloud provide
 Contains configurations for supported cloud providers. At least one provider
 needs to be defined.
 
-- `gcp` [GCP](#gcp)
-  
-  List of configuration options for [Google Cloud](https://cloud.google.com/). This field is optional.
+- `name`
 
-- `hetzner` [Hetzner](#hetzner)
-  
-  List of configuration options for [Hetzner Cloud](https://www.hetzner.com/cloud) . This field is optional.
+  The name of the provider specyfication. It has to be unique across all providers.
 
-- `oci` [OCI](#oci)
-  
-  List of configuration options for [Oracle Cloud Infrastructure](https://www.oracle.com/uk/cloud/). This field is optional.
+- `providerType`
 
-- `aws` [AWS](#aws)
-  
-  List of configuration options for [Amazon Web Services](https://aws.amazon.com/). This field is optional.
+  Type of a provider. The providerType defines mandatory fields that has to be included for a specyfic provider. A list of available providers can be found at [providers section](./providers). Allowed values are:
 
-- `azure` [Azure](#azure)
+  | Value         | Description                              |
+  | ------------- | ---------------------------------------- |
+  | `aws`         | [AWS](#aws) provider type                |
+  | `azure`       | [Azure](#azure) provider type            |
+  | `cloudflare`  | [Cloudflare](#cloudflare) provider type  |
+  | `gcp`         | [GCP](#gcp) provider type                |
+  | `hetzner`     | [Hetzner](#hetzner) provider type        |
+  | `hetznerdns`  | [Hetzner](#hetznerdns) DNS provider type |
+  | `oci`         | [OCI](#oci) provider type                |
   
-  List of configuration options for [Azure](https://azure.microsoft.com/en-gb/). This field is optional.
+- `secretRef` [SecretRef](#secretref)
 
-- `cloudflare` [Cloudflare](#cloudflare)
+  Represents a Secret Reference. It has enough information to retrieve secret in any namespace.
 
-  List of Cloudflare configuration for [Cloudflare](https://www.cloudflare.com/en-gb/). This field is optional.
+Support for more cloud providers is in the [roadmap](https://github.com/berops/claudie/blob/master/docs/roadmap/roadmap.md).
 
-- `hetznerdns` [HetznerDNS](#hetznerdns)
 
-  List of HetznerDNS configuration for [HetznerDNS](https://www.hetzner.com/dns-console). This field is optional.
+## SecretRef
+  
+  SecretReference represents a Kubernetes Secret Reference. It has enough information to retrieve secret in any namespace.
 
-Support for more cloud providers is in the [roadmap](https://github.com/berops/claudie/blob/master/docs/roadmap/roadmap.md).
+- `name`
 
-## Cloudflare
+  Name is unique within a namespace to reference a secret resource.
```

Comment:
`Name of the secret, which holds data for the particular cloud provider instance`